File: 0703-kth-largest-element-in-a-stream/0703-kth-largest-element-in-a-stream.java
class KthLargest {
    PriorityQueue<Integer>p=new PriorityQueue<>();
    int temp;
    public KthLargest(int k, int[] nums) {
    temp=k;
    for(int i : nums)
    {
        p.offer(i);
        if(p.size()>k) p.poll();
    }
    }
    
    public int add(int val) {
    p.offer(val);
    if(p.size()>temp) p.poll();
    return p.peek();
    }
}

/**
 * Your KthLargest object will be instantiated and called as such:
 * KthLargest obj = new KthLargest(k, nums);
 * int param_1 = obj.add(val);
 */

File: 0703-kth-largest-element-in-a-stream/0703-kth-largest-element-in-a-stream.cpp
class KthLargest {
public:

    priority_queue<int , vector<int>, greater<int>>p;
    int temp;
    KthLargest(int k, vector<int>& nums) {
        temp=k;
    for(auto i : nums)
    {
        p.push(i);
        if(p.size()>k) p.pop();
    }
    }
    
    int add(int i) {
        p.push(i);
        if(p.size()>temp) p.pop();
        return p.top();
    }
};

/**
 * Your KthLargest object will be instantiated and called as such:
 * KthLargest* obj = new KthLargest(k, nums);
 * int param_1 = obj->add(val);
 */